File: CodeChef/LongChallenge/January/Doraemon(CHFDORA).c
#include <stdio.h>
int main() {
    int t;
    scanf("%d", &t);
    while (t--) {
        int N, M;
        scanf("%d %d", &N, &M);
        int arr[N][M],i,j;
        for (i = 0; i < N; i++) {
            for (j = 0; j < M; j++) {
                scanf("%d", &arr[i][j]);
            }
        }
        int count=N*M,min1,min2,min3,k;
        for(i=1;i<N-1;i++)
        {
            for(j=1;j<M-1;j++)
            {
                if(N-1-i < M-1-j)
                   min1=N-1-i;
                else
                   min1=M-1-j;

                if(i<j)
                   min2=i;
                else
                   min2=j;

               if(min1 < min2)
                 min3=min1 + 1;
               else
                    min3=min2 + 1;
                for(k=1;k < min3; k++)
                {
                    if(arr[i][j-k]==arr[i][j+k] && arr[i-k][j]==arr[i+k][j])
                        count +=1;
                    else
                        break;
                }
            }
        }
        printf("%d\n",count);
    }
}

File: CodeChef/LongChallenge/January/Doraemon(CHFDORA).java
import java.util.*;
import java.lang.*;
import java.io.*;


public class Main
{
 public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        int t=sc.nextInt();
        while(t-->0)
        {
            int N=sc.nextInt();
            int M=sc.nextInt();
            int [][]arr=new int[N][M];
            for(int i=0;i<N;i++)
            {
                for(int j=0;j<M;j++)
                {
                    arr[i][j]=sc.nextInt();
                }
            }
            int count=N*M;
            for(int i=1;i<N-1;i++)
            {
                for(int j=1;j<M-1;j++)
                {
                    int min1=Math.min(N-1-i,M-1-j);
                    int min2=Math.min(i,j);
                    int min3=Math.min(min1,min2)+1;
                    for(int k=1;k<min3;k++)
                    {
                        if(arr[i][j-k]==arr[i][j+k]&&arr[i-k][j]==arr[i+k][j])
                            count+=1;
                        else
                            break;
                    }
                }
            }
            System.out.println(count);
        }
    }
}
